### ISSUE TITLE
[v0.8.0] Amalgamation broken for Android ARM 64 bit 

### ISSUE THREAD

**Issue by d4wud** [dmlc/mxnet#4554]
Amalgamation does not work out of the box for Android 64 bit ARM architectures.

I finally got it working but I'm not sure I'm doing it right.  Can anyone help?

First I recompiled OpenBLAS and exported a standalone toolchain, both for arm64.  Here are the following changes I had to make:

Use the following values for `CC` and `CXX`:
```
export CC=aarch64-linux-android-gcc   
export CXX=aarch64-linux-android-g++
```
Change the flags in the amalgamation Makefile from 
```
CFLAGS+=  -mhard-float -D_NDK_MATH_NO_SOFTFP=1 -O3
LDFLAGS+=  -Wl,--no-warn-mismatch -lm_hard
```
to:
```
CFLAGS+=  -D_NDK_MATH_NO_SOFTFP=1 -O3 
LDFLAGS+=  -Wl,--no-warn-mismatch
```

QUESTION: -mhard-float and -lm_hard are not supported by the gcc/g++ compilers for arm64.  Is removing them the correct approach, or should they be replaced by something I'm not aware of?

**Comment by piiswrong** [comment]
@antinucleon

**Comment by d4wud** [comment]
@antinucleon I've create an update Makefile that fixes the problem with an extended ARCH argument. Example:
`make ANDROID=1 ARCH=arm64`

I'm happy to update the `Readme.md` and create a pull request?

**Comment by antinucleon** [comment]
Thank you @d4wud 
Recently MXNet upgrade to NNVM. We will create a new amalgamation and please PR to the new one. The new amalgamation is planed to be finish in this weekend.

**Comment by piiswrong** [comment]
This should be fixed for v0.9.3 and later
